Sources/HTTPKit/Client/HTTPClient.swift

```diff
@@ -139,7 +140,13 @@ public final class HTTPClient {
             handlers.append(("client-decoder", clientResDecoder))
             httpHandlerNames.append("client-decoder")
             
-            let clientReqEncoder = HTTPClientRequestEncoder(hostname: hostname)
+            let clientReqEncoder: HTTPClientRequestEncoder
+            if port == defaultSchemePort {
```

Comment:
Could you add a comment pointing to the RFC section for this? That way we can remember why it's important 👍

Reply:
I will! 👍

Reply:
Done